lib/components/ui/confirm-modal.tsx

```diff
@@ -83,7 +82,7 @@ export const ConfirmModal = ({
       <DialogContent className='w-[460px]'>
         <Form {...form}>
           <DialogHeader>
-            <DialogTitle>{title || t('warning')}</DialogTitle>
+            <DialogTitle>{title || context.translations.warning}</DialogTitle>
```

Comment:
Может использование переводов сделать по классике, через t из useTranslation?
Для этого можно в каком-нибудь месте сделать useEffect на изменение переданного в контекст словаря. При изменении мы этот словарь обновляем в i18n и соответственно он автоматом и везде в верстке подхватится.
Или хотел именно отвязаться от i18next?

```suggestion
            <DialogTitle>{title || t('warning')}</DialogTitle>
```

Reply:
1. В целом да, снизится вес библиотеки, но пропадут и фишки: поддержка интерполяции, множественных чисел, дат, вложенных ключей. Если в перспективе они не понадобятся, то можно и оставить чисто контекст.
2. Это всё можно оставить. Я говорил лишь о том, чтобы эти вот переданные в контекст translations помещать в i18next, и далее как обычно использовать i18next в проекте.

Reply:
Если переходить на контекст, то я бы предложил добавить идентичный API для использования переводов.
Создать свой хук useTranslation() и возвращать из него функцию t, которая будет возвращать перевод по ключу из контекста.